packages/contracts-bedrock/src/L1/DataAvailabilityChallenge.sol

```diff
@@ -0,0 +1,266 @@
+// SPDX-License-Identifier: MIT
```

Comment:
Alot of this looks to duplicate the logic from the dispute protocol (challenge status, the rough interface, etc.) - is there any specific reason that this isn't an `IDisputeGame`?

Reply:
Didn't know the `IDisputeGame` interface existed! Agree there is some overlap. But it seems like the `DisputeGameFactory` creates a new `DisputeGame` for every dispute, which is a different approach than this challenge contract. Since challenging a hash and resolving a challenge by providing the pre-image of the hash are both simple atomic operations, they can be managed in a single contract and it would feel like an overkill to create a new `DataAvailabilityDispute` contract for each challenge.